test/es6/proxyconstructor.js

```diff
@@ -0,0 +1,26 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+let arr = new ArrayBuffer(10);
+arr.constructor = new Proxy(ArrayBuffer, {});
+
+arr.slice(1,2);

```

Comment:
Seems like there are two tests here. I'd ad this inside Bugs\misc_bug.js so that we have better description than comments. what do you think?